### PR TITLE
Add a new endpoint to return resource info

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
## Summary

Adding a new endpoint and handler to return some resource information. 

## Test

Will add unit test

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java [hunk 1]
```diff
@@ -2162,4 +2177,134 @@ static class ResourceProperty {
...
+    String dcName = clusterMapConfig.clusterMapDatacenterName;
+    List<String> liveInstances = new ArrayList<>(getLiveInstanceCount(resourceName));
+    List<String> unavailableInstances = new ArrayList<>(getUnavailableInstanceCount(resourceName));
+    long totalCapacity = getResourceTotalRegisteredHostDiskCapacity(resourceName);
+    long liveCapacity = getResourceAvailableRegisteredHostDiskCapacity(resourceName);
+    long unavailableCapacity = totalCapacity - liveCapacity;
+    int numPartitions = getNumberOfPartitionsInResource(resourceName);
+    int replicationFactor = 3; // by default it is 3;
+    String numReplicaStr = getResourceConfig(resourceName, dcName).getNumReplica();
+    if (numReplicaStr != null) {
```
Comment (JingQianCloud) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java:2289]: We may hit exception if the string is not a number.

**Review by JingQianCloud — APPROVED** [review]
LGTM. And thanks for adding another tooling for the helix clustermap.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java [hunk 2]
```diff
@@ -2162,4 +2177,134 @@ static class ResourceProperty {
...
+
+    /**
+     * Return the resource name that the given partition belongs to.
+     * @param helixClusterManager
+     * @return
+     */
+    @Override
+    public List<String> identifyResources(HelixClusterManager helixClusterManager) {
+      Map<String, Set<String>> partitionToResourceMap =
+          helixClusterManager.partitionToResourceNameByDc.get(helixClusterManager.clusterMapConfig);
```
Comment (Arun-LinkedIn) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java:2215]: Probably `helixClusterManager.clusterMapConfig.clusterMapDatacenterName`?

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java [hunk 3]
```diff
@@ -2162,4 +2177,134 @@ static class ResourceProperty {
...
+    }
+
+    /**
+     * Check if the given resource is valid, and then return it back.
+     * @param helixClusterManager
+     * @return
+     */
+    @Override
+    public List<String> identifyResources(HelixClusterManager helixClusterManager) {
+      if (!helixClusterManager.dcToResourceNameToTag.get(helixClusterManager.clusterMapConfig).contains(resourceName)) {
```
Comment (Arun-LinkedIn) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java:2244]: same as above.. `helixClusterManager.clusterMapConfig.clusterMapDatacenterName`?

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java [hunk 4]
```diff
@@ -2162,4 +2177,134 @@ static class ResourceProperty {
...
+
+    /**
+     * Return all the resource names this cluster map has.
+     * @param helixClusterManager
+     * @return
+     */
+    @Override
+    public List<String> identifyResources(HelixClusterManager helixClusterManager) {
+      return new ArrayList<>(
+          helixClusterManager.dcToResourceNameToTag.get(helixClusterManager.clusterMapConfig).keySet());
```
Comment (Arun-LinkedIn) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java:2265]: Same as above..

**Review by Arun-LinkedIn — APPROVED** [review]
Thank you for adding this endpoint to get resource information. Just had one comment for typo.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java]: updated, my bad.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java:2290]: then we will get an exception and return 503 back to client

**Pull request merged**